flaml/automl/task/generic_task.py

```diff
@@ -125,7 +180,7 @@ def validate_data(
                     "object",
                     "string",
                 ), "If the task is an NLP task, X can only contain text columns"
-                for each_cell in X[column]:
+                for _, each_cell in X[column].items():
```

Comment:
@liususan091219 could you confirm this change is OK?

Reply:
not sure why we need this change?

Reply:
yes